torch/fx/passes/tensorify_python_scalars.py

```diff
@@ -0,0 +1,245 @@
+            return sympy_interp(Analysis, expr_to_tensor_proxy, expr)  # type: ignore[arg-type]
+
+        # hash cons on arguments, run expr handler
+        expr_to_tensor_proxy[expr] = _run_sympy_handler(
+            Analysis,
+            [_sympy_interp(arg) for arg in expr.args],  # type: ignore[arg-type]
+            expr,
+        )
+
+        return expr_to_tensor_proxy[expr]
+
+    nodes = list(graph.nodes)
+    for i, node in enumerate(nodes[:-1]):
+        with graph.inserting_before(
+            nodes[i + 1] if node not in placeholders else first_non_placeholder
+        ):
+            # Look for tensor.item() calls on placeholders
+            if unbacked_bindings := node.meta.get("unbacked_bindings"):
+                for s in unbacked_bindings.keys():
+                    if (
+                        node is not None
+                        and node.op == "call_function"
+                        and node.target is torch.ops.aten._local_scalar_dense.default
+                    ):
+                        dtype = node.args[0].meta["val"].dtype
+                        if dtype != torch.float64:
+                            continue
+
+                        assert isinstance(node.args[0], fx.Node), node.args[0]
+
+                        expr_to_tensor_proxy[s] = fx.Proxy(node.args[0], tracer=tracer)
+                        expr_to_sym_proxy[s] = fx.Proxy(node, tracer=tracer)
+
+            elif (sym_expr := _get_sym_val(node)) is not None:
+                if sym_expr not in expr_to_sym_proxy and not isinstance(
+                    sym_expr, (sympy.Number, sympy.logic.boolalg.BooleanAtom)
+                ):
+                    expr_to_sym_proxy[sym_expr] = fx.Proxy(node, tracer=tracer)
+
+            # Look for functions to convert
+            if node.op == "call_function" and node.target is torch.ops.aten.mul.Tensor:
+                args = []
+                transform = False
+                for a in node.args:
+                    if isinstance(a, fx.Node) and isinstance(
+                        zf := a.meta["val"], torch.SymFloat
+                    ):
+                        transform = True
+                        try:
+                            interp_node = _sympy_interp(zf.node.expr).node
+                        except NotImplementedError:
+                            transform = False
+                            break
+
+                        # Upcast to higher float32 precision during computation if dtype is float16. Later
+                        # on we will downcast back to float16 after the computation is done. Checkout the
+                        # elementwise_dtypes function for more context:
+                        # https://github.com/pytorch/pytorch/blob/main/torch/_prims_common/__init__.py#L1379
+                        compute_dtype = (
+                            torch.float32
+                            if node.meta["val"].dtype == torch.float16
```

Comment:
You can't only do this for float16, this is needed also for bfloat16, among others. Check elementwise_dtypes to see if there's a way you can do it without duplicating all the logic here.